src/main/java/net/coderbot/iris/mixin/MixinAbstractBlock.java

```diff
@@ -0,0 +1,31 @@
+package net.coderbot.iris.mixin;
+
+import net.coderbot.iris.block_rendering.BlockRenderingSettings;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.minecraft.block.AbstractBlock;
+import net.minecraft.block.BlockState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.BlockView;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+
+@Mixin(AbstractBlock.class)
+public class MixinAbstractBlock {
+	/**
+	 * @author IMS
+	 * @reason ambientOcclusionLevel support
+	 */
+	@Environment(EnvType.CLIENT)
+	@Deprecated
+	@Overwrite
+	public float getAmbientOcclusionLightLevel(BlockState state, BlockView world, BlockPos pos) {
```

Comment:
This doesn't correctly implement the function described in https://github.com/IrisShaders/ShaderDoc/blob/master/const-directives.md#ambient-occlusion-level

Also, if something overrides this function, the resulting value won't be modified properly. It's probably safer to mix in to BlockState / AbstractBlockState instead.